### PR TITLE
Categorize column choices

### LINKED ISSUE #958: Streamline & categorise column choices
- names
- descriptions
- logos
- categories

### PR DESCRIPTION
**Pull request by richardolsson**
## Description
This PR implements categorization of column choices, adds some new choices and improves some text.

## Screenshots
image

## Changes
* Refactors column choices code
  * Separates code into multiple files for easier browsing and reading
  * Switches to using an object map instead of an array for choices
* Separates choices into categories
* Adds some shortcut field choices for creating email and phone fields
* Adds some prefabricated smart search queries
* Fixes a bug causing columns to be created in the wrong order (mentioned in #965)

## Notes to reviewer
Please feedback on the texts!

I made some changes compared to what we agreed on the issue:

* Instead of having "first name" and "last name" alongside "full name", I made it so that the "full name" option will be enabled if either field is missing, and add the missing one(s)
* Instead of putting "email" and "phone" under "Outreach", I put the "Pick field" choice so that the user can pick any field with just a single choice

## Related issues
Resolves #958

### CODE REVIEW

**Comment by ziggabyte** [comment]
Input from looking at and thinking about the names and feel of the dialog: 
Add a short description to email and phone number (just because it looks too empty otherwise)

Custom x-ref description: "See if each person appears in a Smart Search query."
Reached on phone description: "Have we ever reached them when phonebanking?"
Survey response + Survey Responses - rename to "One question" and "Many questions"

File: src/locale/misc/views/en.yml [hunk 1]
```diff
@@ -19,31 +34,62 @@ columnDialog:
...
+      columnTitle: Participated
+      description: Were they ever booked on an event?
+      keywords: active booked
+      title: Ever participated
+    queryReached:
+      columnTitle: Reached
+      description: Have we ever reached when phonebanking?
+      keywords: phone
+      title: Reached on phone
     surveyResponse:
```
Comment (ziggabyte) [inline src/locale/misc/views/en.yml:93]: I put my comments about texts in a separate comment above!

File: src/features/views/components/ViewColumnDialog/ColumnGallery/SearchResults.tsx [hunk 2]
```diff
@@ -33,9 +33,9 @@ const SearchResults = ({
             <ColumnChoiceCard
               alreadyInView={alreadyInView}
               cardVisual={searchResult.renderCardVisual(
-                alreadyInView ? 'gray' : '#234890'
+                alreadyInView ? 'gray' : searchResult.color
               )}
-              color={alreadyInView ? 'gray' : '#234890'}
+              color={alreadyInView ? 'gray' : searchResult.color}
```
Comment (ziggabyte) [inline src/features/views/components/ViewColumnDialog/ColumnGallery/SearchResults.tsx:38]: This is my mistake, for not saying when asking you to do this: the "gray" should be replaced by "theme.palette.onSurface.disabled"

**Review by ziggabyte — CHANGES_REQUESTED** [review]
what a nice refactor of the choices and categories! I ask for a small refactor but that is actually my own fault, so apologies for that 😅

**Comment by rebecarubio** [comment]
> ponses - should be different

Super good job categorizing all the choices, it looks brilliant and it´s more understandable✨ ! I agree with all that @ziggabyte had suggested, I could only add that we have talked to change color and icon for the `Survey Responses` choice, to be the same as `Delegate` 
image

**Review by ziggabyte — CHANGES_REQUESTED** [review]
A few detail comments, open for discussion on all of them. 

This does highlight the faults in the logic we created for colors (blue - 1 column, purple - 2 columns, red - multi columns) since there are choices that add 1, 2 or more columns. I think it's fine that there are inconsistensies in this now, it needs more dedicated thought at a separate time.

**Review by ziggabyte — APPROVED** [review]
LET'S GOOOOOO!!!

**Pull request merged**